src/main/api/2/studio-api-2.yaml

```diff
@@ -5522,6 +5522,167 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /workflow/item_states:
+    get:
+      tags:
+        - workflow
+      summary: Get item workflow states for site
+      description: 'Required permission "get_item_states"'
+      operationId: getItemStates
+      parameters:
+        - name: siteId
+          in: query
+          description: site ID
+          required: true
+          schema:
+            type: string
+        - name: path
+          in: query
+          description: path regex
+          required: false
+          schema:
+            type: string
+        - name: states
+          in: query
+          description: states bitmap mask
+          required: false
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/SandboxItem'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+
+    post:
+      tags:
+        - workflow
+      summary: Set item workflow states for site
+      description: 'Required permission "set_item_states"'
+      operationId: setItemStates
+      requestBody:
+        description: List of items to set workflow state(s)
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId: 
+                  type: string
+                  description: site ID
+                items:
+                  type: array
+                  description: path(s) of item(s)
+                  items:
+                    type: string
+                clearSystemProcessing:
+                  type: boolean
+                  description: true if system processing is to be cleared
+                clearUserLocked:
+                  type: boolean
+                  description: true if user lock is to be cleared
+                live:
+                  type: boolean
+                  description: true if item is to be set as live, otherwise false
+                staged:
+                  type: boolean
+                  description: true if item is to be set as staged, otherwise false
```

Comment:
- Aren't we missing state here too? (i.e. to set the item(s) state)?
- Are all these JSON attributes optional? For example, we may want to just set one of them with out altering the rest.

Reply:
@dejan-brkic please make the bits optional.

Reply:
@rart I think we got it all, ping me and I can explain better.